token/program-2022/src/extension.rs

```diff
@@ -38,60 +303,218 @@ pub enum Extension {
     /// Includes an optional mint close authority
     MintCloseAuthority,
 }
+impl TryFrom<&[u8]> for ExtensionType {
+    type Error = ProgramError;
+    fn try_from(a: &[u8]) -> Result<Self, Self::Error> {
+        Self::try_from(u16::from_le_bytes(
+            a.try_into().map_err(|_| ProgramError::InvalidAccountData)?,
+        ))
+        .map_err(|_| ProgramError::InvalidAccountData)
+    }
+}
+impl From<ExtensionType> for [u8; 2] {
+    fn from(a: ExtensionType) -> Self {
+        u16::from(a).to_le_bytes()
+    }
+}
+impl ExtensionType {
+    /// Get the data length of the type associated with the enum
+    pub fn get_associated_type_len(&self) -> usize {
+        match self {
+            ExtensionType::Uninitialized => 0,
+            ExtensionType::MintTransferFee => pod_get_packed_len::<MintTransferFee>(),
+            ExtensionType::AccountTransferFee => pod_get_packed_len::<AccountTransferFee>(),
+            ExtensionType::MintCloseAuthority => pod_get_packed_len::<MintCloseAuthority>(),
+        }
+    }
+}
 
-/// Type-Length-Value Entry, used to encapsulate all extensions contained within an account
-#[repr(C)]
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct TlvEntry<V> {
-    /// Extension type encoded in the rest of the entry
-    pub extension: Extension,
-    /// Length of the entry, in bytes
-    pub length: u32,
-    /// Deserialized value
-    pub value: V,
+/// Get the required account data length for the given ExtensionTypes
+pub fn get_account_len(extension_types: &[ExtensionType]) -> usize {
+    let extension_size: usize = extension_types
+        .iter()
+        .map(|e| {
+            e.get_associated_type_len()
+                .saturating_add(size_of::<ExtensionType>())
+                .saturating_add(pod_get_packed_len::<Length>())
+        })
+        .sum();
+    let total_extension_size = if extension_size == Multisig::LEN {
+        extension_size + 1
```

Comment:
Comment explaining this magic number?
Actually, I'm a little confused by this, as I though that we'd be using `ExtensionType::Uninitialized` (2 bytes) for this case.

Reply:
Oh yes, very good find! you're absolutely right